### PR TITLE
Add changelog for 4.0.0-beta.1

### PR DESCRIPTION
**Pull request by consideRatio**
## Omitted PRs

<details><summary>Click to expand</summary>

- Bump kubespawner from 6.2.0 to 7.0.0b1 [#3520](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3520) ([@consideRatio](https://github.com/consideRatio))
- Bump oauthenticator from 16.3.1 to 17.0.0 [#3519](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3519) ([@consideRatio](https://github.com/consideRatio))
- Update jupyterhub/configurable-http-proxy version from 4.6.1 to 4.6.2 [#3431](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3431) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@minrk](https://github.com/minrk))
- Update library/traefik version from v3.1.3 to v3.1.4 [#3504](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3504) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@manics](https://github.com/manics))
- Update library/traefik version from v3.1.2 to v3.1.3 [#3503](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3503) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@consideRatio](https://github.com/consideRatio))
- Update library/traefik version from v3.1.1 to v3.1.2 [#3477](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3477) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@manics](https://github.com/manics))
- Update library/traefik version from v3.1.0 to v3.1.1 [#3469](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3469) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@manics](https://github.com/manics))
- Update library/traefik version from v3.0.4 to v3.1.0 [#3460](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3460) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@manics](https://github.com/manics))
- Update library/traefik version from v3.0.3 to v3.0.4 [#3451](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3451) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@minrk](https://github.com/minrk))
- Update library/traefik version from v3.0.2 to v3.0.3 [#3438](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3438) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@minrk](https://github.com/minrk))
- Update library/traefik version from v3.0.1 to v3.0.2 [#3432](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3432) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@manics](https://github.com/manics))
- Update library/traefik version from v3.0.0 to v3.0.1 [#3421](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3421) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@manics](https://github.com/manics))
- Update library/traefik version from v2.11.2 to v3.0.0 [#3410](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3410) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@consideRatio](https://github.com/consideRatio))
- Update library/traefik version from v2.11.0 to v2.11.2 [#3393](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3393) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@consideRatio](https://github.com/consideRatio))
- Update kube-scheduler version from v1.28.13 to v1.28.14 [#3496](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3496) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@consideRatio](https://github.com/consideRatio))
- Update kube-scheduler version from v1.28.12 to v1.28.13 [#3482](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3482) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@minrk](https://github.com/minrk))
- Update kube-scheduler version from v1.28.11 to v1.28.12 [#3462](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3462) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@consideRatio](https://github.com/consideRatio))
- Update kube-scheduler version from v1.28.10 to v1.28.11 [#3433](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3433) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@manics](https://github.com/manics))
- Update kube-scheduler version from v1.28.9 to v1.28.10 [#3417](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3417) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@manics](https://github.com/manics))
- Update kube-scheduler version from v1.28.8 to v1.28.9 [#3399](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3399) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@consideRatio](https://github.com/consideRatio))
- Update jupyterhub from 5.0.0 to 5.1.0 [#3472](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3472) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@consideRatio](https://github.com/consideRatio))
- Update jupyterhub from 5.0.0b1 to 5.0.0b2 [#3416](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3416) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@minrk](https://github.com/minrk))
- Update jupyterhub from 5.0.0b2 to 5.0.0 [#3425](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3425) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@consideRatio](https://github.com/consideRatio))
- Update jupyterhub from 4.1.5 to 4.1.6 [#3471](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3471) ([@consideRatio](https://github.com/consideRatio))
- Update jupyterhub from 4.1.4 to 4.1.5 [#3390](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3390) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@consideRatio](https://github.com/consideRatio))
- Update jupyterhub from 4.1.3 to 4.1.4 [#3384](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3384) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@consideRatio](https://github.com/consideRatio))
- Update jupyterhub from 4.1.2 to 4.1.3 [#3381](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3381) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@consideRatio](https://github.com/consideRatio))
- Update jupyterhub from 4.1.1 to 4.1.2 [#3378](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3378) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@consideRatio](https://github.com/consideRatio))
- Update jupyterhub from 4.1.0 to 4.1.1 [#3375](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3375) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@consideRatio](https://github.com/consideRatio))
- Update pause version from 3.9 to 3.10 [#3423](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3423) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@manics](https://github.com/manics))
- Add changelog for 3.3.8 [#3473](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3473) ([@consideRatio](https://github.com/consideRatio), [@manics](https://github.com/manics))
- Add changelog for 3.3.7 [#3392](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3392) ([@consideRatio](https://github.com/consideRatio))
- Add changelog for 3.3.7 [#3391](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3391) ([@consideRatio](https://github.com/consideRatio))
- Add changelog for 3.3.6 [#3385](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3385) ([@consideRatio](https://github.com/consideRatio))
- Add changelog for 3.3.5 [#3382](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3382) ([@consideRatio](https://github.com/consideRatio))
- Add changelog for 3.3.4 [#3379](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3379) ([@consideRatio](https://github.com/consideRatio))
- Add changelog for 3.3.3 [#3376](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3376) ([@consideRatio](https://github.com/consideRatio), [@minrk](https://github.com/minrk))
- Add changelog for 3.3.2 [#3370](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3370) ([@consideRatio](https://github.com/consideRatio))
- Add changelog for 3.3.1 [#3367](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3367) ([@consideRatio](https://github.com/consideRatio))
- Bump peter-evans/create-pull-request from 6 to 7 [#3524](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3524) ([@consideRatio](https://github.com/consideRatio))
- Bump aquasecurity/trivy-action from 0.23.0 to 0.24.0 [#3474](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3474) ([@minrk](https://github.com/minrk))
- Bump aquasecurity/trivy-action from 0.21.0 to 0.23.0 [#3449](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3449) ([@manics](https://github.com/manics))
- Bump aquasecurity/trivy-action from 0.19.0 to 0.21.0 [#3428](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3428) ([@manics](https://github.com/manics))
- build(deps): bump aquasecurity/trivy-action from 0.18.0 to 0.19.0 [#3388](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3388) ([@consideRatio](https://github.com/consideRatio))
- Bump oauthenticator from 16.3.0 to 16.3.1 in /images/hub [#3518](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3518) ([@minrk](https://github.com/minrk))
- Bump cryptography from 43.0.0 to 43.0.1 in /images/singleuser-sample [#3491](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3491) ([@manics](https://github.com/manics))
- Bump jupyterlab from 4.2.4 to 4.2.5 in /images/singleuser-sample [#3486](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3486) ([@manics](https://github.com/manics))
- Bump aiohttp from 3.10.0 to 3.10.2 in /images/hub [#3479](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3479) ([@manics](https://github.com/manics))
- Bump certifi from 2024.6.2 to 2024.7.4 in /images/singleuser-sample [#3453](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3453) ([@manics](https://github.com/manics))
- Bump certifi from 2024.6.2 to 2024.7.4 in /images/hub [#3452](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3452) ([@manics](https://github.com/manics))
- Bump urllib3 from 2.2.1 to 2.2.2 in /images/hub [#3445](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3445) ([@manics](https://github.com/manics))
- Bump github.com/hashicorp/go-retryablehttp from 0.7.1 to 0.7.7 in /images/image-awaiter [#3444](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3444) ([@manics](https://github.com/manics))
- Bump urllib3 from 2.2.1 to 2.2.2 in /images/singleuser-sample [#3437](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3437) ([@manics](https://github.com/manics))
- Bump tornado from 6.4 to 6.4.1 in /images/singleuser-sample [#3430](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3430) ([@manics](https://github.com/manics))
- Bump tornado from 6.4 to 6.4.1 in /images/hub [#3429](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3429) ([@manics](https://github.com/manics))
- build(deps): bump pymysql from 1.1.0 to 1.1.1 in /images/hub [#3420](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3420) ([@manics](https://github.com/manics))
- hub image: refreeze requirements.txt [#3525](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3525) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@consideRatio](https://github.com/consideRatio))
- hub image: refreeze requirements.txt [#3516](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3516) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@consideRatio](https://github.com/consideRatio))
- hub image: refreeze requirements.txt [#3456](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3456) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@consideRatio](https://github.com/consideRatio))
- hub image: refreeze requirements.txt [#3435](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3435) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@consideRatio](https://github.com/consideRatio))
- hub image: refreeze requirements.txt [#3434](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3434) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@minrk](https://github.com/minrk))
- hub image: refreeze requirements.txt [#3414](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3414) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@minrk](https://github.com/minrk))
- hub image: refreeze requirements.txt [#3409](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3409) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@minrk](https://github.com/minrk))
- hub image: refreeze requirements.txt [#3408](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3408) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@consideRatio](https://github.com/consideRatio))
- hub image: refreeze requirements.txt [#3402](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3402) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@consideRatio](https://github.com/consideRatio))
- Vulnerability patch in singleuser-sample [#3500](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3500) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@manics](https://github.com/manics))
- Vulnerability patch in hub [#3499](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3499) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@manics](https://github.com/manics))
- Vulnerability patch in network-tools [#3498](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3498) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@manics](https://github.com/manics))
- Vulnerability patch in hub [#3495](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3495) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@manics](https://github.com/manics))
- Vulnerability patch in secret-sync [#3494](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3494) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@minrk](https://github.com/minrk))
- Vulnerability patch in singleuser-sample [#3489](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3489) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@manics](https://github.com/manics))
- Vulnerability patch in hub [#3488](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3488) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@manics](https://github.com/manics))
- Vulnerability patch in hub [#3480](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3480) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@minrk](https://github.com/minrk))
- Vulnerability patch in network-tools [#3475](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3475) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@manics](https://github.com/manics))
- Vulnerability patch in singleuser-sample [#3467](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3467) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@manics](https://github.com/manics))
- Vulnerability patch in hub [#3466](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3466) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@manics](https://github.com/manics))
- Vulnerability patch in secret-sync [#3454](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3454) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@manics](https://github.com/manics))
- Vulnerability patch in singleuser-sample [#3448](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3448) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@manics](https://github.com/manics))
- Vulnerability patch in hub [#3447](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3447) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@manics](https://github.com/manics))
- Vulnerability patch in network-tools [#3446](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3446) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@manics](https://github.com/manics))
- Vulnerability patch in secret-sync [#3443](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3443) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@minrk](https://github.com/minrk))
- Vulnerability patch in secret-sync [#3436](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3436) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@manics](https://github.com/manics))
- Vulnerability patch in secret-sync [#3397](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3397) ([@jupyterhub-bot](https://github.com/jupyterhub-bot), [@consideRatio](https://github.com/consideRatio))
- Vulnerability patch in network-tools [#1947](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/1947) ([@github-actions](https://github.com/github-actions))
- [pre-commit.ci] pre-commit autoupdate [#3476](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3476) ([@minrk](https://github.com/minrk))
- [pre-commit.ci] pre-commit autoupdate [#3450](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3450) ([@minrk](https://github.com/minrk))
- [pre-commit.ci] pre-commit autoupdate [#3413](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3413) ([@consideRatio](https://github.com/consideRatio))
- [pre-commit.ci] pre-commit autoupdate [#3389](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3389) ([@consideRatio](https://github.com/consideRatio))

</details>

### CODE REVIEW

**Comment by consideRatio** [comment]
@manics the changelog isn't final yet, but what remains now is the declaring the date and populating linked upgrade docs that is currently just a placeholder page.

File: docs/source/changelog.md [hunk 1]
```diff
@@ -12,16 +12,102 @@ changes in pull requests], this list should be updated.
...
+## 4.0
 
-[kubespawner's changelog]: https://jupyterhub-kubespawner.readthedocs.io/en/stable/changelog.html
+### 4.0.0-beta.1 - 2024-10-XX
+
+This release updates JupyterHub itself from version 4 to 5, and the dependencies
+`jupyterhub-kubespawner`, `oauthenticator`, and `ldapauthenticator` to a new
+major version. Please read the summary of breaking changes below and the
+[documentation on upgrading from chart version 3 to 4](upgrade-3-to-4) before
+upgrading.
```
Comment (manics) [inline docs/source/changelog.md:23]:
```suggestion
major version.

We will provide an [upgrade guide for 3 to 4](upgrade-3-to-4) before the final release, but for now please read the summary of breaking changes below, and the linked changelogs.
```

**Review by manics — COMMENTED** [review]
An upgrade guide is needed for the final release, but if there are no other breaking changes or major bugs I think it's fine to release the beta now.

**Comment by manics** [comment]
CI linkcheck error: `/home/runner/work/zero-to-jupyterhub-k8s/zero-to-jupyterhub-k8s/docs/source/changelog.md.md:447: WARNING: Duplicate reference definition: IMAGES/HUB/REQUIREMENTS.TXT`

**Comment by consideRatio** [comment]
Updated release date, fixed link, added most recently merged PRs, rebased and squashed commits. Let's do a release!

**Pull request merged**